src/ai/defaultai.cc

```diff
@@ -6237,14 +6406,18 @@ void DefaultAI::consider_productionsite_influence(BuildableField& field,
 	for (const Widelands::DescriptionIndex& i : bo.ware_outputs) {
 		++field.producers_nearby.at(i);
 	}
-	if (bo.has_collected_map_resource()) {
-		++field.collecting_producers_nearby.at(bo.get_collected_map_resource());
+	for (const auto& supported_building : bo.supported_producers) {
+		// Using std::map syntax to create entry on demand.
+		++field.supported_producers_nearby[supported_building.first];
 	}
 
-	if (!bo.production_hints.empty()) {
-		for (auto ph : bo.production_hints) {
-			++field.supporters_nearby.at(ph);
+	const Widelands::ProductionSiteDescr* productionsite =
+	   dynamic_cast<const Widelands::ProductionSiteDescr*>(bo.desc);
+	for (const auto& supported : productionsite->supported_productionsites()) {
+		if (field.supporters_nearby.count(supported) != 1) {
+			field.supporters_nearby[supported] = 0;
 		}
+		++field.supporters_nearby[supported];
```

Comment:
same here as above. we need to count the building found at this coordinate not the buildings supported by this one.

Reply:
forget this one as here it is correct we count the building that was found as supporter for each building it supports